newIDE/app/src/VariablesList/index.js

```diff
@@ -173,23 +194,45 @@ export default class VariablesList extends React.Component<Props, State> {
         variable={variable}
         disabled={depth !== 0}
         depth={depth}
+        type={type}
         errorText={
           this.state.nameErrors[variable.ptr]
             ? 'This name is already taken'
             : undefined
         }
         onChangeValue={text => {
-          variable.setString(text);
+          // if it's an object variable edited, create an instance variable in it's place marked as 'inherited'
+          if (inheritedVariablesContainer && type === 'object') {
+            const serializedVariable = serializeToJSObject(
+              inheritedVariablesContainer.get(name)
+            );
+            const newVariable = new gd.Variable();
+            unserializeFromJSObject(newVariable, serializedVariable);
+            variablesContainer.insert(name, newVariable, index);
+            newVariable.delete();
+
+            variablesContainer.get(name).setString(text);
+          } else {
+            variable.setString(text);
+          }
+
           this.forceUpdate();
           if (this.props.onSizeUpdated) this.props.onSizeUpdated();
         }}
+        onResetToDefaultValue={() => {
+          if (type === 'inherited') {
```

Comment:
type will always be inherited right?

Reply:
yes it will, but the flow test was failing for some reason without this

Reply:
ok it works now :D